### PR TITLE
Spec for Drill-down navigation in grafana dashboard

### PR DESCRIPTION
**Pull request by cloudbehl**
tendrl-bug-id: https://github.com/Tendrl/specifications/issues/189
Signed-off-by: cloudbehl cloudbehl@gmail.com

### CODE REVIEW

**Comment by cloudbehl** [comment]
A visual depiction of how drill-down workflow will be.
download

@anivargi @brainfunked @r0h4n @anmolbabu

**Review by anivargi — APPROVED** [review]

**Comment by julienlim** [comment]
@cloudbehl  @anivargi @brainfunked @r0h4n @anmolbabu @japplewhite 

In looking at the visual depiction of the workflow, there appears to be some items that are gaps that are not addressed:
- active snapshots counter
- Geo-rep sessions counter
- Active Geo-rep sessions counter
- heal (healinfo) counter
- Split brain files with issues counter
- rebalancing counter (if applicable)
- Volumes status/health should include Degraded (and not just Up and Down)
- Connections counter (to bricks and volumes)
- active alerts counter

Is the diagram supposed to be exhaustive or just a placeholder to get started?

**Comment by cloudbehl** [comment]
@julienlim It's just a place holder to get started. It is how we are planning to start implementing the drill downs in grafana dashboard.

File: specs/grafana-drill-down.adoc [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+None
+
+=== Data model impact:
+
+==== Graphite Data model needs to be in below-mentioned structure for allowing users to drill-down.
+  Volume and Brick Dashboard
+  collectd.clusters."cluster_id".volumes."volume_name".hosts."host_name".bricks."brick_path".*(All the Volume and Brick level data)
+
+  Host level Dashboard:
+  collecd.cluster."cluster_id".nodes."host_name".*(All the node level data)
```
Comment (cloudbehl) [inline specs/grafana-drill-down.adoc:50]: @nthomas-redhat these are the queries which are created to achieve the drill down in grafana.

File: specs/grafana-drill-down.adoc [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+== Proposed change
+
+
+The structure of graphite is going to be changed for allowing user to 
+context switch between dashboards.
+
+=== The new structure for graphite will be:
+  Volume and Brick Dashboard
+  collectd.clusters."cluster_id".volumes."volume_name".hosts."host_name".bricks."brick_path".*(All the volume level data)
```
Comment (cloudbehl) [inline specs/grafana-drill-down.adoc:31]: @nthomas-redhat the "cluster_id", "volume_name" , "host_name" and "brick_path" are the variables that are going to be dynamically populated to create a graphite query

**Comment by r0h4n** [comment]
@cloudbehl @anivargi How much of this is ready for Milestone2? and what is left to be done for Milestone 3?

**Comment by cloudbehl** [comment]
@r0h4n Drill down is not implemented in milestone 2, all feature related to drill-down will be targeted for implementation in milestone 3.

**Pull request merged**